### ISSUE TITLE
[Telegram bot API bug] Wring UTF-8 detection in bot keyboards

### ISSUE THREAD

**Issue by halaei** [telegramdesktop/tdesktop#2452]
When I send a message with inline keyboard containing string "￿^_^ Ŋ  ɨოﾑ Sʊŋ￿^_^" UTF-8 encoded as: "\xEF\xBF\xBF\x5E\x5F\x5E\x20\xC5\x8A\x20\x20\xC9\xA8\xE1\x83\x9D\xEF\xBE\x91\x20\x53\xCA\x8A\xC5\x8B\xEF\xBF\xBF\x5E\x5F\x5E" I recieve this error:
"Bad Request: Keyboard button text must be encoded in UTF-8"
But according to all the tests I have done, including the using standart php function [mb_check_encoding](http://php.net/manual/en/function.mb-check-encoding.php) this string is a truely valid UTF-8 encoded string.

So in short why do you think "\xEF\xBF\xBF" is not a valid UTF-8 character? I guess you are 100% wrong.

**Comment by john-preston** [comment]
@halaei Maybe this chars crash iOS or smth? I don't know, that is not a Telegram Desktop report, but something about the server side.

**Comment by halaei** [comment]
I did a research, it sounds like you should exclude \xEF\xBF\xBD not \xEF\xBF\xBF. From wiki:

> The replacement character � (often a black diamond with a white question mark) is a symbol found in the Unicode standard at codepoint U+FFFD in the Specials table. It is used to indicate problems when a system is not able to decode a stream of data to a correct symbol. It is most commonly seen when a font does not contain a character, but is also seen when the data is invalid and does not match any character:

I know this is not a desktop issue, but could your team please provide a list of characters that are valid UTF-8 but should not be used in keyboard? I think there are other invalid characters as well. I contacted bot support, no progress yet. :(

Thanks

**Comment by halaei** [comment]
Sorry I don't know about that quote. Please ignore the quote. But an answer still would be great.

**Comment by john-preston** [comment]
@halaei And what is the use case for such buttons?

**Comment by halaei** [comment]
I only put first_name of users into inline buttons. It is a game @minroobot

**Comment by tdesktop-closer** [comment]
Hey there!

We're automatically closing this issue since there was no activity in this issue since 411 days ago. We therefore assume that the user has lost interest or resolved the problem on their own. Closed issues that remain inactive for a long period may get automatically locked.

Don't worry though; if this is in error, let us know with a comment and we'll be happy to reopen the issue.

Thanks!

<sub>(Please note that this is an [automated](https://github.com/auchri/github_auto_closer) comment.)</sub>
